lib/src/builder/transformer_builder.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:async';
+
+import 'package:barback/barback.dart' as barback;
+
+import '../asset/id.dart';
+import '../builder/builder.dart';
+import '../util/barback.dart';
+import 'build_step.dart';
+
+/// A builder which wraps a [barback.DeclaringTransformer].
+///
+/// **Note:** Not all [barback.Transform] apis are supported, for instance files
+/// may only be read as [String]s currently.
+class TransformerBuilder implements Builder {
+  final barback.Transformer transformer;
+
+  TransformerBuilder(this.transformer) {
+    /// Unfortunately, we can't add a static type for this :(.
+    if (transformer is! barback.DeclaringTransformer) {
+      throw 'Only DeclaringTransformers are supported, which $transformer is '
```

Comment:
Create an `Exception` or `Error` class for this. `ArgumentError` might be a good fit, actually